src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -332,13 +335,85 @@ public override IntermediateNode Visit(TexlCallNode node, IRTranslatorContext co
                     }
                 }
 
+                // This can add pre-processing to arguments, such as BlankToZero, Truncate etc...
+                // based on the function.
+                args = AttachArgPreprocessor(args, func);
+
                 // this can rewrite the entire call node to any intermediate node.
                 // e.g. For Boolean(true), Instead of IR as Call(Boolean, true) it can be rewritten directly to emit true.
                 var irNode = func.CreateIRCallNode(node, context, args, scope);
 
                 return MaybeInjectCoercion(node, irNode, context);
             }
 
+            private List<IntermediateNode> AttachArgPreprocessor(List<IntermediateNode> args, TexlFunction func)
+            {
+                var len = args.Count;
+                List<IntermediateNode> convertedArgs = new List<IntermediateNode>(len);
+
+                for (var i = 0; i < len; i++)
+                {
+                    IntermediateNode convertedNode = default;
+                    var argPreprocessor = func.GetArgPreprocessor(i);
+
+                    switch (argPreprocessor)
+                    {
+                        case ArgPreprocessor.ReplaceWithZero:
+                            convertedNode = ReplaceBlankWithZero(args[i]);
+                            break;
+                        default:
+                            convertedNode = args[i];
+                            break;
+                    }
+
+                    convertedArgs.Add(convertedNode);
+                }
+
+                return convertedArgs;
+            }
+
+            /// <summary>
+            /// Wraps node arg => Coalesce(arg , 0) when arg is not Number Literal.
+            /// </summary>
+            private static IntermediateNode ReplaceBlankWithZero(IntermediateNode arg)
+            {
+                if (arg is NumberLiteralNode)
+                {
+                    return arg;
+                }
+
```

Comment:
@MikeStall Double checking if this is enough for Dataverse case that we talked offline,